src/main/java/randoop/resource/generator/MethodListGen.java

```diff
@@ -0,0 +1,252 @@
+                  new EverythingAllowedPredicate());
+              sideEffectMethodWriter.write(fullyQualifiedMethodSignature);
+              sideEffectMethodWriter.newLine();
+            } catch (RandoopUsageError e) {
+              System.err.println("Not parsable: " + e.getMessage());
+              unparsableSideEffectMethodWriter.write(fullyQualifiedMethodSignature);
+              unparsableSideEffectMethodWriter.newLine();
+            }
+          }
+          sideEffectMethodWriter.flush();
+        }
+      } else {
+        System.out.println("No side effect free methods found. Not writing to file.");
+      }
+    } catch (IOException e) {
+      e.printStackTrace();
+      System.exit(1);
+    }
+  }
+
+  /**
+   * Returns the annotated methods from the the given jar.
+   *
+   * @param jarFile jdk jar
+   * @param annotationCategory type of annotation we are parsing for
+   * @param qualifyingAnnotations list of annotations
+   * @return annotated methods in fully-qualified signature format
+   */
+  private static List<String> getAnnotatedMethodsFromJDKJar(
+      Path jarFile,
+      AnnotationCategory annotationCategory,
+      Collection<String> qualifyingAnnotations) {
+    List<String> annotatedMethods = new ArrayList<>();
+    try {
+      JarFile jar = new JarFile(jarFile.toFile());
+
+      // Read through each entry in a jar via stream
+      Stream<JarEntry> jarEntries = jar.stream();
+      jarEntries.forEach(
+          jarEntry -> {
+            try {
+              if (jarEntry.getName().endsWith(CLASS_EXT)) {
+                InputStream is = jar.getInputStream(jarEntry);
+                annotatedMethods.addAll(
+                    readClassFile(is, annotationCategory, qualifyingAnnotations));
+              }
+            } catch (IOException e) {
+              throw new RuntimeException("Failure to parse: " + e.getMessage());
+            }
+          });
+    } catch (IOException e) {
+      throw new RuntimeException("Failure to parse: " + e.getMessage());
+    }
+
+    // Sort in alphabetical order
+    Collections.sort(annotatedMethods);
+    return annotatedMethods;
+  }
+
+  /**
+   * Reads a class file using the given input stream and parses it for the desired annotations.
+   *
+   * @param classInputStream input stream used for reading
+   * @param annotationCategory type of annotation we are parsing for
+   * @param qualifyingAnnotations list of annotations
+   * @return list of methods with the desired annotations in fully qualified signature format
+   * @throws IOException if SceneLib fails to parse the class file
+   */
+  private static List<String> readClassFile(
+      InputStream classInputStream,
+      AnnotationCategory annotationCategory,
+      Collection<String> qualifyingAnnotations)
+      throws IOException {
+    List<String> annotatedMethods = new ArrayList<>();
+
+    // Invoke the parser for the specified class file
+    AScene scene = new AScene();
```

Comment:
The scene library is poor code that we are trying to get away from.  I would like to not add new dependencies on it.  Furthermore, I don't want to make the Checker Framework a dependency of Randoop.

There are better ways to parse a .class file.  The most popular library is ASM, but Randoop already uses BCEL, so I suggest that you use it.  That avoids adding a dependency.  Mark is an expert on BCEL in case you need any advice.